src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendVideoBookingService.kt

```diff
@@ -0,0 +1,96 @@
+  private val videoBookingRepository: VideoBookingRepository,
+  prisonAppointmentRepository: PrisonAppointmentRepository,
+  prisonRepository: PrisonRepository,
+  locationValidator: LocationValidator,
+  prisonerValidator: PrisonerValidator,
+) : CreateVideoAppointmentService(prisonRepository, prisonAppointmentRepository, locationValidator, prisonerValidator) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  fun amend(videoBookingId: Long, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val booking = videoBookingRepository.findById(videoBookingId)
+      .orElseThrow { EntityNotFoundException("Video booking with ID $videoBookingId not found.") }
+      .also {
+        require(BookingType.valueOf(it.bookingType) == request.bookingType) {
+          "The booking type ${it.bookingType} does not match the requested type ${request.bookingType}."
+        }
+      }
+
+    return when (BookingType.valueOf(booking.bookingType)) {
+      BookingType.COURT -> amendCourt(booking, request, amendedBy)
+      BookingType.PROBATION -> amendProbation(booking, request, amendedBy)
+    }
+  }
+
+  private fun amendCourt(booking: VideoBooking, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val court = courtRepository.findByCode(request.courtCode!!)
+      ?.also { require(it.enabled) { "Court with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Court with code ${request.courtCode} not found")
+
+    val prisoner = request.prisoner().validate()
+
+    booking.apply {
+      this.court = court
+      hearingType = request.courtHearingType!!.name
+      comments = request.comments
+      videoUrl = request.videoLinkUrl
+      this.amendedBy = amendedBy
+      amendedTime = LocalDateTime.now()
+    }
+
+    return booking.let(videoBookingRepository::saveAndFlush)
+      .also { deletePrisonAppointments(it) }
+      .also { createAppointmentsForCourt(booking, request.prisoner()) }
+      .also { log.info("BOOKINGS: court booking ${it.videoBookingId} amended") } to prisoner
+  }
+
+  private fun amendProbation(booking: VideoBooking, request: AmendVideoBookingRequest, amendedBy: String): Pair<VideoBooking, Prisoner> {
+    val probationTeam = probationTeamRepository.findByCode(request.probationTeamCode!!)
+      ?.also { require(it.enabled) { "Probation team with code ${it.code} is not enabled" } }
+      ?: throw EntityNotFoundException("Probation team with code ${request.probationTeamCode} not found")
+
+    val prisoner = request.prisoner().validate()
+
+    booking.apply {
+      this.probationTeam = probationTeam
+      probationMeetingType = request.probationMeetingType!!.name
+      comments = request.comments
+      videoUrl = request.videoLinkUrl
+      this.amendedBy = amendedBy
+      amendedTime = LocalDateTime.now()
+    }
+
```

Comment:
I'm working on a booking history table now, so will probably add a line or two here when ready. It records the current, new state of the key details of the booking - enough so that we can find and replace the external appointments that related to the previous state.